src/libAtoms/MPI_context.f95

```diff
@@ -1517,10 +1533,109 @@ subroutine MPI_context_scatterv_int1(this, v_in, v_out, counts, root, error)
 
   call MPI_scatterv(v_in, counts, displs, MPI_INTEGER, v_out, count, MPI_INTEGER, my_root, this%communicator, err)
   PASS_MPI_ERROR(err, error)
+
+  if (allocated(displs)) deallocate(displs)
 #endif
 
 end subroutine MPI_context_scatterv_int1
 
+subroutine MPI_context_scatterv_real1(this, v_in, v_out, counts, root, error)
+  type(MPI_context), intent(in) :: this
+  real(dp), intent(in) :: v_in(:)
+  real(dp), intent(out) :: v_out(:)
+  integer, intent(in) :: counts(:)
+  integer, intent(in), optional :: root
+  integer, intent(out), optional :: error
+
+  integer :: my_root, err, i, count
+  integer, allocatable :: displs(:)
+
+  INIT_ERROR(error)
+
+  if (.not. this%active) then
+    if (any(shape(v_in) /= shape(v_out))) then
+      RAISE_ERROR("MPI_context_scatterv_real1 (no MPI) shape mismatch: v_in " // shape(v_in) // " v_out " // shape(v_out), error)
+    endif
+    v_out = v_in
+    return
+  endif
+
+#ifdef _MPI
+  my_root = optional_default(ROOT_, root)
+
+  call mpi_scatter(counts, 1, MPI_INTEGER, count, 1, MPI_INTEGER, my_root, this%communicator, err)
+  PASS_MPI_ERROR(err, error)
+
+  if (count > size(v_out)) then
+    RAISE_ERROR("MPI_context_scatterv_real1 not enough space: count " // count // " size(v_out) " // size(v_out), error)
+  endif
+
+  if (is_root(this, my_root)) then
+    allocate(displs(this%n_procs))
+    displs(1) = 0
+    do i = 2, this%n_procs
+      displs(i) = displs(i-1) + counts(i-1)
+    end do
```

Comment:
In the other subroutine, there is an `else` for the non-root processes, but not here. Maybe there is an explanation for this asymmetry? What do non-root processes do with an unallocated `displs` array?

Reply:
If have added the else cases so that `-fcheck-all` does not make the `test_gapfit.py` fail.